src/TimeSpans.jl

```diff
@@ -84,15 +87,22 @@ function translate(span, by::Period)
     return TimeSpan(start(span) + by, stop(span) + by)
 end
 
+# Extend the Base function when defined. TODO: Remove this when 1.5 is required.
+if isdefined(Base, :contains)  # VERSION >= v"1.5.0-DEV.639"
+    import Base: contains
+end
+
 """
-    TimeSpans.contains(a, b)
+    contains(a, b)
 
 Return `true` if the timespan `b` lies entirely within the timespan `a`, return `false` otherwise.
 """
-contains(a, b) = start(a) <= start(b) && stop(a) >= stop(b)
```

Comment:
This `contains` intentionally does not extend `Base.contains`, so we probably actually should not export it (instead calls to it should be qualified). The TimeSpans interface is purposefully duck-typed so we want to keep the original generic definition here.

Reply:
> This `contains` intentionally does not extend `Base.contains`

Okay, thanks for the context. What was the reasoning behind that design decision?

Reply:
> What was the reasoning behind that design decision?

The arguments to it are not type constrained, so we can't overload `Base.contains`.

Are you do mean more generally "why provide a duck-typed interface?" That's just so that people don't have to subtype a specific supertype for TimeSpan-like functionality